include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -1,322 +1,118 @@
-    Cube(std::array<std::shared_ptr<Cube>, 8> &octants, float size, const glm::vec3 &position);
-
-    Cube(const Cube &cube);
-
-    Cube(Cube &&cube) noexcept;
-
-    Cube &operator=(Cube &&lhs) noexcept;
-
-    Cube &operator=(const Cube &rhs);
-
-    /// Parse an octree from binary data.
-    /// @param data The data to parse the octree from.
-    /// @return Cube object representing the cubes / octrees from the data.
-    static Cube parse(std::vector<unsigned char> &data);
-
-    /// Parse an octree from a BitStream.
-    /// @param stream The BitStream to parse the octree from.
-    /// @return Cube object representing the cubes / octrees from the stream.
-    static Cube parse(BitStream &stream);
-
-    /// Parse an octree from a BitStream.
-    /// @param stream The BitStream to parse the octree from.
-    /// @param size The maximum size of the cube.
-    /// @param position The position of the cube in the coordinate system (i.e., the vector from (0, 0, 0) to the bounds
-    /// of the cube with the lowest values on x, y, and z-axis).
-    /// @return Cube object representing the cubes / octrees from the stream.
-    static Cube parse(BitStream &stream, float size, const glm::vec3 &position);
-
-    /// Get the type of the cube.
-    /// @return type of the cube.
-    [[nodiscard]] CubeType type();
-
-    /// Get the number of leaves, this octree contains.
-    /// Leaves are cubes of CubeType::INDENTED or CubeTYPE::FULL.
-    /// @return Number of leaves, this octree contains.
-    [[nodiscard]] std::uint64_t leaves();
-
-    // [[nodiscard]] dynamic_bitset<> bits(); // Bit representation of this cube
-    // [[nodiscard]] vector<array<glm::vec3, 8>> vertices(); // All vertices this cube contains.
-    // [[nodiscard]] vector<array<glm::vec3, 4>> sides(); // All sides this cube contains.
-    /// Get all polygons (triangles) of each cube of this octree.
-    /// @return A vector which contains the three vertices representing a triangle.
-    [[nodiscard]] std::vector<std::array<glm::vec3, 3>> polygons(); // All polygons this cube contains.
-
-    /// Invalidate the cache of this cube / octree (not its children).
-    void invalidate_cache();
-
-    /// Make this octree reactive (update its values when one of its attributes changes).
-    /// @param force Whether to make it reactive again even though the connections were established before.
-    void make_reactive(bool force = false);
+    Cube() = default;
+    explicit Cube(const Type type);
+    Cube(const Type type, const float size, const glm::vec3 &position);
+    Cube(std::weak_ptr<Cube> parent, const Type type, const float size, const glm::vec3 &position);
+    Cube(const Cube &rhs);
+    Cube(Cube &&rhs) noexcept;
+    ~Cube() = default;
+    Cube &operator=(Cube rhs);
+    /// Get child.
+    std::shared_ptr<Cube> operator[](const std::size_t idx);
+    /// Get child.
+    const std::shared_ptr<const Cube> operator[](const std::size_t idx) const;
```

Comment:
```suggestion
    std::shared_ptr<Cube> operator[](std::size_t idx);
    /// Get child.
    const std::shared_ptr<const Cube> operator[](std::size_t idx) const;
```

Reply:
Fixed in d81c9e9576c42a3b00b817bb4727c44dc02ff2d6